internal/store/ingress.go

```diff
@@ -104,10 +104,18 @@ var (
 			GenerateFunc: wrapIngressFunc(func(i *v1beta1.Ingress) *metric.Family {
 				ms := []*metric.Metric{}
 				for _, rule := range i.Spec.Rules {
-					for _, path := range rule.HTTP.Paths {
+					if rule.HTTP != nil {
+						for _, path := range rule.HTTP.Paths {
+							ms = append(ms, &metric.Metric{
+								LabelKeys:   []string{"host", "path", "service_name", "service_port"},
+								LabelValues: []string{rule.Host, path.Path, path.Backend.ServiceName, path.Backend.ServicePort.String()},
+								Value:       1,
+							})
+						}
+					} else {
 						ms = append(ms, &metric.Metric{
 							LabelKeys:   []string{"host", "path", "service_name", "service_port"},
-							LabelValues: []string{rule.Host, path.Path, path.Backend.ServiceName, path.Backend.ServicePort.String()},
+							LabelValues: []string{rule.Host, "", "", ""},
```

Comment:
If there is no path, I don't think this metric should be exposed at all I think, so there shouldn't be the else path. Any reason we would need it? (remember, that labels with empty values are just thrown away by Prometheus)

Reply:
I'm good with not exposing this metric. Will make the change.